python-spec/src/somacore/coordinates.py

```diff
@@ -0,0 +1,521 @@
+"""Definitions of types related to coordinate systems."""
+
+import abc
+import collections.abc
+from dataclasses import dataclass
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+
+
+@dataclass
+class Axis(metaclass=abc.ABCMeta):
+    """A description of an axis of a coordinate system
+
+    Args:
+        name: Name of the axis.
+        unit: Optional string units. Defaults to ``None``.
+        scale: Optional scale for units. Defaults to ``None``.
+
+    Lifecycle: experimental
+    """
+
+    name: str
+    units: Optional[str] = None
```

Comment:
```suggestion
    unit: Optional[str] = None
```

Should this be singular, `unit`?

Reply:
It is singular in the doc string